src/main/java/pl/lodz/p/java/flyweight/repository/BudgetRepository.java

```diff
@@ -0,0 +1,11 @@
+package pl.lodz.p.java.flyweight.repository;
+
+import java.util.List;
+import pl.lodz.p.java.flyweight.model.Budget;
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+public interface BudgetRepository extends MongoRepository<Budget, String>,  BudgetRepositoryCustom {
```

Comment:
Obawiałam się tego pytania :). Opierałam sie na pewnym kodzie, który nie jest moją właśnoscią.  Zwróć uwagę że 
BudgetRepository extends MongoRepository<Budget, String>,  BudgetRepositoryCustom
public class BudgetRepositoryImpl implements MongoCustom, BudgetRepositoryCustom

Przyznam się szczerze że nie doczytałam co sie kryje pod spodem, ale domyślam się że jest w tym kolejna magiczna sztuczka Springa

Jak próbowałam połączyć to w jedno to musiałabym zaiplementować metody MongoRepository a tak nie muszę